service/collector/chart_resource.go

```diff
@@ -80,6 +91,21 @@ func (c *ChartResource) Collect(ch chan<- prometheus.Metric) error {
 			key.ReleaseStatus(chartConfig),
 			key.Namespace(chartConfig),
 		)
+
+		if key.IsCordoned(chartConfig) {
```

Comment:
We could have an early continue and prevent unnecessary indentations.

Reply:
This is the earliest checkpoint for `IsCordoned` function could be called before we sent another metrics. Can you suggest which line I can move it to?

Reply:
I mean something like this. 

```
if !key.IsCordoned(cr) {
  continue
}

// the other code here
```